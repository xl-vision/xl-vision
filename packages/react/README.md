# `@xl-vision/react`

> TODO: description

