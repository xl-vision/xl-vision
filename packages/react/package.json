{
  "name": "@xl-vision/react",
  "version": "0.0.0",
  "description": "xl-vision react components.",
  "author": "Rhys Xia <xrs4433@outlook.com>",
  "license": "MIT",
  "main": "legacy/index.js",
  "module": "modern/index.js",
  "types": "legacy/index.d.ts",
  "files": [
    "legacy",
    "modern"
  ],
  "keywords": [
    "xl-vision",
    "react",
    "styles"
  ],
  "homepage": "https://github.com/xl-vision/xl-vision-next/tree/master/packages/react#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/xl-vision/xl-vision-next.git"
  },
  "scripts": {
    "build:legacy": "node ../../scripts/build legacy",
    "build:modern": "node ../../scripts/build modern",
    "build": "npm run build:legacy && npm run build:modern"
  },
  "bugs": {
    "url": "https://github.com/xl-vision/xl-vision-next/issues"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.5",
    "prop-types": "^15.7.2"
  },
  "devDependencies": {
    "@types/react": "^16.8.0",
    "@types/react-dom": "^16.8.0",
    "@xl-vision/styled-engine": "^0.0.0",
    "react": "^16.8.0",
    "react-dom": "^16.8.0"
  },
  "peerDependencies": {
    "@types/react": "^16.8.0 || ^17.0.0",
    "@xl-vision/styled-engine": "^0.0.0",
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  },
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  }
}
