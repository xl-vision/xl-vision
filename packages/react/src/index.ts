export { styled, createGlobalStyles } from './styles';
export { default as ThemeProvider } from './ThemeProvider';
export { default as Transition } from './Transition';
export { default as CSSTransition } from './CSSTransition';
export { default as CollapseTransition } from './CollapseTransition';
export { default as TransitionGroup } from './TransitionGroup';
