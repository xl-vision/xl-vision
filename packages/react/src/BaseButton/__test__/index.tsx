import { mount } from 'enzyme';
import { BaseButton } from '@xl-vision/react';

describe('BaseButton', () => {
  it('basic test', () => {
    const handleClick = jest.fn();

    const wrapper = mount(<BaseButton onClick={handleClick}>button</BaseButton>);

    wrapper.find(BaseButton).simulate('click');

    expect(handleClick.mock.calls.length).toBe(1);

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('测试disabled', () => {
    const handleClick = jest.fn();

    const wrapper = mount(
      <BaseButton disabled={true} onClick={handleClick}>
        button
      </BaseButton>,
    );

    wrapper.find(BaseButton).simulate('click');
    wrapper.update();

    expect(handleClick.mock.calls.length).toBe(0);
  });

  it('测试loading', () => {
    const handleClick = jest.fn();

    const wrapper = mount(
      <BaseButton loading={true} onClick={handleClick}>
        button
      </BaseButton>,
    );

    wrapper.find(BaseButton).simulate('click');
    wrapper.update();

    expect(handleClick.mock.calls.length).toBe(0);
  });
});
