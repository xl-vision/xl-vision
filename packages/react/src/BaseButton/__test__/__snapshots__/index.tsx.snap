// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton Test basic render 1`] = `
.c2 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
  overflow: hidden;
  border-radius: inherit;
  pointer-events: none;
}

.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  background-color: transparent;
  border: 0;
  border-radius: 0;
  outline: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0 svg {
  pointer-events: none;
}

.c0 .xl-base-button__ripple {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 0.12;
}

.c0 .xl-base-button__ripple-enter-active {
  -webkit-transition: all 225ms 0ms cubic-bezier(0,0,0.2,1);
  transition: all 225ms 0ms cubic-bezier(0,0,0.2,1);
}

.c0 .xl-base-button__ripple-exit-active {
  -webkit-transition: all 195ms 0ms cubic-bezier(0.4,0,1,1);
  transition: all 195ms 0ms cubic-bezier(0.4,0,1,1);
}

.c0 .xl-base-button__ripple-enter-from {
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 0.1;
}

.c0 .xl-base-button__ripple-exit-to {
  opacity: 0;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 100%;
  height: 100%;
}

<div>
  <button
    class="c0 xl-base-button"
  >
    <span
      class="c1 xl-base-button__inner"
    >
      button
    </span>
    <div
      class="c2 xl-ripple xl-base-button__ripple"
    />
  </button>
</div>
`;
